### ISSUE TITLE
Type error when attempting to iterate through AxiosHeaders class

### ISSUE THREAD

**Issue by podjames** [axios/axios#5452]
### Describe the bug

Axios 1.22 added `Symbol.iterator` implementation for the AxiosHeaders class. 

However attempting to iterate through the headers class (for example - using `Object.fromEntries(headers)`) results in a type error.

It looks like `Symbol.iterator` has been implemented in the code for the AxiosHeaders class but has been missed in the Typescript definitions.

### Code snippet

```js
const headers = new AxiosHeaders({foo: "bar"})

for (const header of headers) {
    console.log(header);
}

error TS2488: Type 'AxiosHeaders' must have a '[Symbol.iterator]()' method that returns an iterator.
```

### Expected behavior

AxiosHeaders class should not throw a type error when attempting to iterate through its properties.

**Comment by bobvandevijver** [comment]
Probably related to https://github.com/axios/axios/issues/5416.

**Comment by podjames** [comment]
Looks like this has been fixed since v1.2.4 - thanks! Will close this 👍
